### DISCUSSION TITLE
Variable to see if Himalaya is active

### DISCUSSION THREAD

**Discussion by dennis-n-schneider** [pimalaya/himalaya#410] [Ideas]
In order to write plugin-local key bindings, I would like to know if there is a variable I can use to check if Himalaya is currently opened in this nvim session. So far I couldn't find anything on this, but for custom keybindings this would be very important. (Compare e.g. to coc's "coc#pum#visible" which checks if the popup-menu is currently visible, resulting in keybindings only triggering when actually using the plugin).

**Comment by soywod** [comment]
What do you mean by "to check if Himalaya is currently opened in this nvim session"? If one of the listing/reading/writing buffer is open?

**Reply by dennis-n-schneider** [reply]
Yes, that is exactly what I meant. I couldn't find anything on this.

**Reply by soywod** [reply]
I guess it is doable, but I have hard time to understand how useful it could be. The comparison with `coc#pum#visible` is not fair because the popup-menu works only inside buffers, and it makes sense to have different keybindings depending on the state of this popup-menu. But Himalaya only uses buffers, keybindings should remain with those buffers and should not be accessible from outside. It is even dangerous IMO? Can you tell me a bit more about the use case?

**Reply by dennis-n-schneider** [reply]
Maybe my approach is wrong, but the idea was to set a keybinding to e.g. "Enter" (Enter might be a bit inappropriate, but some key-binding) which opens the mail-message and starts goyo and limelight for better readability. However, I obviously only want this keybinding to be active when actually in the window using Himalaya and not in my normal nvim session.

**Reply by soywod** [reply]
I understand better. This is related to your specific workflow, not Himalaya, so I would suggest you to write your own code and eventually share it on the wiki, in the Tips section. It should be doable using `:ls` to list current buffers, check if you find a buffer starting by `Himalaya`, load it and then start goyo or whatever.

**Reply by dennis-n-schneider** [reply]
Ah, I see! Thank you very much and sorry for the confusion
